content/environment_variables.tex

```diff
@@ -30,14 +30,31 @@
     \tabularnewline\hline
 %%
 \EnvVarDecl{SHMEM\_SYMMETRIC\_SIZE}
-    & Non-negative integer
-    & Number of bytes to allocate for symmetric heap
+    & Positive decimal value (integer or floating point) with an optional
+    character suffix %that specifies a scaling factor
+    & The value set in this environment variable is interpreted as an
+    implementation-defined size (in bytes) at least as large as the integer
+    ceiling of the product of the numeric prefix and the scaling factor. The
+    allowed character suffix for the scaling factor are as follows:
+      \begin{itemize}
+        \item k or K multiplies by \(2^{10}\)  (kilobytes)
+        \item m or M multiplies by \(2^{20}\)  (megabytes)
+        \item g or G multiplies by \(2^{30}\)  (gigabytes)
+        \item t or T multiplies by \(2^{40}\)  (terabytes)
+      \end{itemize}
+      For example, string \enquote{20m} is equivalent to the integer value
+      20971520, or 20 megabytes. Similarly the string \enquote{3.1m} is
+      equivalent to the integer value 3250586. Only one multiplier is
+      recognized, so \enquote{20kk} will not produce the same desired result as
+      \enquote{20m}. Usage of string \enquote{.5m} will yield the same result as
+      the string \enquote{0.5m}. Any unrecognized suffix is treated as an error
+      and results in program termination with an error report.
```

Comment:
@nspark made a comment that `shmem_init_thread` could return (an unspecified) non-zero status instead of aborting. Assuming the error report is left in, this request is reasonable. Perhaps something like:
> Any unrecognized suffix is an error, generates an error report, and either returns from `shmem_init_thread` or otherwise causes program termination.

The second clause is left vague, but it only affects `shmem_init` today. Writing that in explicitly is also an option:

> Any unrecognized suffix is an error, generates an error report, and either returns from `shmem_init_thread` or causes program termination from `shmem_init`.

The **Return Values** from `shmem_init_thread` already says it returns non-zero if not success.

There's the separate concern of whether this error should be consistent and always result in program termination. (This would be my preference, but others can chime in.)

Reply:
There have been (and may still be) SHMEM implementations that map the physical address space during loading and, as such, will detect this error before any user code runs.  The first text suggestion may better support such implementations by allowing them to abort before any OpenSHMEM library calls are performed (including `shmem_init_thread`).

Reply:
I too would prefer the first text. Will make the change and do a special ballot. Let me know, if this works.